pkgs/top-level/perl-packages.nix

```
@@ -15282,6 +15282,9 @@ with self;
       url = "mirror://cpan/authors/id/X/XA/XAOC/Gtk2-1.24993.tar.gz";
       hash = "sha256-ScRDdDsu7+EadoACck9/akxI78lP8806VZ+357aTyWc=";
     };
+
+    env.NIX_CFLAGS_COMPILE ="-std=gnu17";
```

```suggestion
    env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isGNU "-std=gnu17";
```

This will break clang somehow (or `optionalAttrs` may be better)?

---

Sorry, wrong suggestion. You can build on clang to see whether it works.

---

I guess doing it your suggested way with optionalAttrs like I've done now would ensure it doesn't affect clang at least?  I see there's some other examples of that pattern as well in perlpackages.